hivemq-edge/src/frontend/src/extensions/datahub/components/controls/PolicyToolbar.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC, useMemo, useState } from 'react'
+import { NodeProps, useReactFlow } from 'reactflow'
+import { useTranslation } from 'react-i18next'
+import { Button, ButtonGroup, Card, CardFooter, Icon, useToast, UseToastOptions } from '@chakra-ui/react'
+import { MdOutlineDeleteForever, MdOutlineSave } from 'react-icons/md'
+
+import { ProblemDetailsExtended } from '@/api/types/http-problem-details.ts'
+
+import useDataHubDraftStore from '@datahub/hooks/useDataHubDraftStore.ts'
+import PolicyErrorReport from '@datahub/components/helpers/PolicyErrorReport.tsx'
+import { BehaviorPolicyData, DataHubNodeData, DataPolicyData, PolicyDryRunStatus } from '@datahub/types.ts'
+import { usePolicyDryRun } from '@datahub/hooks/usePolicyDryRun.ts'
+import { getDryRunStatusIcon } from '@datahub/utils/node.utils.ts'
+
+interface ActionPolicyCheckProps {
+  node: NodeProps<DataPolicyData | BehaviorPolicyData>
+}
+
+export const PolicyToolbar: FC<ActionPolicyCheckProps> = (props) => {
+  const { t } = useTranslation('datahub')
+  const [state, setState] = useState<PolicyDryRunStatus>(PolicyDryRunStatus.IDLE)
+  const store = useDataHubDraftStore()
+  const { nodes, onUpdateNodes } = store
+  const createToast = useToast()
+  const { fitView } = useReactFlow()
+  const { checkPolicyAsync } = usePolicyDryRun()
+
+  // only allow a single check at any given time
+  const POLICY_CHECK_ID = 'POLICY_CHECK_ID'
+
+  const TOAST_DEFAULTS: UseToastOptions = {
```

Comment:
suggestion: nice to have toast that are scoped to DataHub.

I think we can even move this one into as separated file that lives inside of the module to encourage reusage within it. WDYT?

Reply:
done